app/services/courses/generate_course_title_service.rb

```diff
@@ -0,0 +1,54 @@
+module Courses
+  class GenerateCourseTitleService
+    def execute(course:)
+      subjects = course.subjects
+
+      title = if course.further_education_course?
+                further_education_title
+              elsif is_modern_lanuage_course?(subjects)
+                modern_language_title(subjects)
+              else
+                generated_title(subjects)
+              end
+
+      title = append_send_info(title) if course.is_send?
+      title
+    end
+
+  private
+
+    def is_modern_lanuage_course?(subjects)
+      subjects[0] == SecondarySubject.modern_languages
+    end
+
+    def modern_language_title(subjects)
+      title = "Modern Languages"
+
+      if subjects.length == 2
```

Comment:
Are we testing if there is more than one subject here? Or language?

Reply:
It feels to me like we should be testing for how many languages there are on the course. I'm guessing we don't have an existing scope for that, but it'd be nice to have. Likewise, we could have a scope for the different types of subjects: `primary_subjects`, `secondary_subjects`, `languages`, etc.

Reply:
Yeah, I was thinking something similar at the time - would be nice to be able to query the languages on the course. I'll give it a look - I wasn't a great fan of it looking at subjects like ^ but wanted to get this in front of people for the general logic anyway